src/fu-udev-device.c

```diff
@@ -133,27 +145,34 @@ fu_udev_device_probe (FuDevice *device, GError **error)
 		return TRUE;
 
 	/* set ven:dev:rev */
-	priv->vendor = fu_udev_device_get_sysfs_attr_as_uint64 (priv->udev_device, "vendor");
-	priv->model = fu_udev_device_get_sysfs_attr_as_uint64 (priv->udev_device, "device");
-	priv->revision = fu_udev_device_get_sysfs_attr_as_uint64 (priv->udev_device, "revision");
+	priv->vendor = fu_udev_device_get_sysfs_attr_as_uint32 (priv->udev_device, "vendor");
+	priv->model = fu_udev_device_get_sysfs_attr_as_uint32 (priv->udev_device, "device");
+	priv->revision = fu_udev_device_get_sysfs_attr_as_uint8 (priv->udev_device, "revision");
 
 	/* fallback to the parent */
 	udev_parent = g_udev_device_get_parent (priv->udev_device);
 	if (udev_parent != NULL &&
 	    priv->vendor == 0x0 && priv->model == 0x0 && priv->revision == 0x0) {
-		priv->vendor = fu_udev_device_get_sysfs_attr_as_uint64 (udev_parent, "vendor");
-		priv->model = fu_udev_device_get_sysfs_attr_as_uint64 (udev_parent, "device");
-		priv->revision = fu_udev_device_get_sysfs_attr_as_uint64 (udev_parent, "revision");
+		priv->vendor = fu_udev_device_get_sysfs_attr_as_uint32 (udev_parent, "vendor");
+		priv->model = fu_udev_device_get_sysfs_attr_as_uint32 (udev_parent, "device");
+		priv->revision = fu_udev_device_get_sysfs_attr_as_uint8 (udev_parent, "revision");
 	}
 
 	/* hidraw helpfully encodes the information in a different place */
 	if (udev_parent != NULL &&
 	    priv->vendor == 0x0 && priv->model == 0x0 && priv->revision == 0x0 &&
 	    g_strcmp0 (priv->subsystem, "hidraw") == 0) {
 		tmp = g_udev_device_get_property (udev_parent, "HID_ID");
-		if (tmp != NULL && strlen (tmp) == 22) {
-			priv->vendor = fu_udev_device_read_uint16 (tmp + 9);
-			priv->model = fu_udev_device_read_uint16 (tmp + 18);
+		if (tmp != NULL) {
+			g_auto(GStrv) split = g_strsplit (tmp, ":", -1);
+			if (g_strv_length (split) == 3) {
+				guint64 val = g_ascii_strtoull (split[1], NULL, 16);
+				if (val < G_MAXUINT32)
+					priv->vendor = val;
```

Comment:
I think we want a g_warning here, right?

Reply:
for the inverse; yes. i'll add.